File: 4344/4344.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   4344.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/09/27 14:37:24 by jihoolee          #+#    #+#             */
/*   Updated: 2021/09/27 14:37:39 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <iomanip>
#include <vector>

using namespace std;

float	get_over_avg(vector<int> scores)
{
	int		over_avg = 0;
	float	avg = 0.;

	for (int i = 0; i < scores.size(); i++)
		avg += scores[i];
	avg /= scores.size();
	for (int i =0; i < scores.size(); i++)
		if (scores[i] > avg)
			over_avg++;
	return (float(over_avg) / (float)scores.size() * 100.);
}

int	main(void)
{
	int						classes_num;
	int						students_num;
	int						temp;
	cin >> classes_num;
	vector<vector<int> >	classes_scores(classes_num);

	for (int i = 0; i < classes_num; i++)
	{
		cin >> students_num;
		for (int j = 0; j < students_num; j++)
		{
			cin >> temp;
			classes_scores[i].push_back(temp);
		}
	}
	cout << fixed;
	cout.precision(3);
	for (int i = 0; i < classes_num; i++)
		cout << get_over_avg(classes_scores[i]) << '%' << endl;
	return (0);
}

File: 4344/4344.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    4344.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/09/27 14:37:48 by jihoolee          #+#    #+#              #
#    Updated: 2021/09/27 14:37:50 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_over_avg(scores_lst):
	over_avg_num = 0
	avg = sum(scores_lst) / len(scores_lst)
	for score in scores_lst:
		if (score > avg): over_avg_num += 1
	return (over_avg_num / len(scores_lst))

length = int(input())
classes_lst = [list(map(int, input().split(' '))) for _ in range(length)]

for scores_lst in classes_lst:
	scores_lst.pop(0)

over_avg_lst = [get_over_avg(scores_lst) for scores_lst in classes_lst]
for over_avg_rate in over_avg_lst:
	print('{0:.3f}%'.format(over_avg_rate * 100))
